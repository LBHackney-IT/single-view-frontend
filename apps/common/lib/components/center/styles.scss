.mtfh-center {
  display: flex;

  &--vertical {
    align-items: center;
  }

  &--horizontal {
    justify-content: center;
  }
}
