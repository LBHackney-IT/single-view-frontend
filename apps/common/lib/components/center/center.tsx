import React, { forwardRef } from "react";

import cn from "classnames";

import { widthOverrides } from "../../utils";

import type * as Polymorphic from "@radix-ui/react-polymorphic";
import "./styles.scss";

export interface CenterProps {
  horizontally?: boolean;
  vertically?: boolean;
  override?: number;
}

export type CenterComponent = Polymorphic.ForwardRefComponent<"div", CenterProps>;

export const Center: CenterComponent = forwardRef(function Center(
  {
    as: CenterComp = "div",
    horizontally = true,
    vertically = true,
    className,
    override,
    ...props
  },
  ref,
) {
  return (
    <CenterComp
      ref={ref}
      className={cn(
        "mtfh-center",
        {
          "mtfh-center--horizontal": horizontally,
          "mtfh-center--vertical": vertically,
        },
        widthOverrides(override),
        className,
      )}
      {...props}
    />
  );
});
