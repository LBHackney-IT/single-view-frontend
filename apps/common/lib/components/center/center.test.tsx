import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { Center } from "./center";

test("it renders with both vertical and horizontal by default", async () => {
  const { container } = render(
    <Center>
      <div />
    </Center>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders with vertical only", () => {
  const { container } = render(
    <Center horizontally={false}>
      <div />
    </Center>,
  );
  expect(container).toMatchSnapshot();
});

test("it renders with horizontal only", () => {
  const { container } = render(
    <Center vertically={false}>
      <div />
    </Center>,
  );
  expect(container).toMatchSnapshot();
});
