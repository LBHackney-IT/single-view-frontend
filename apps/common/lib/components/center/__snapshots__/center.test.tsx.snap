// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders with both vertical and horizontal by default 1`] = `
<div>
  <div
    class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
  >
    <div />
  </div>
</div>
`;

exports[`it renders with horizontal only 1`] = `
<div>
  <div
    class="mtfh-center mtfh-center--horizontal"
  >
    <div />
  </div>
</div>
`;

exports[`it renders with vertical only 1`] = `
<div>
  <div
    class="mtfh-center mtfh-center--vertical"
  >
    <div />
  </div>
</div>
`;
